### PR TITLE
Port static turn annotations to Hydra

### PR DESCRIPTION
**Pull request by EricMichaelSmith**
- Converts static turn annotations Mephisto task to use Hydra
- Adds the ability to softblock MTurk workers in any Mephisto task in `parlai.crowdsourcing`
- Standardizes file names across Mephisto tasks
- Creates central README explaining how to use Mephisto tasks with Hydra

**Testing**
```
# Base task (no in-flight QA):
python parlai/crowdsourcing/tasks/turn_annotations_static/run.py \
mephisto.blueprint.data_jsonl=/checkpoint/ems/2020_multimodal_blender/q_function/for_fast_acute/results_20200904_153005_live_dialogpt.json \
mephisto.task.task_name=turn_annotations_static_foo2

# In-flight QA task:
python parlai/crowdsourcing/tasks/turn_annotations_static/run_in_flight_qa.py \
mephisto.blueprint.data_jsonl=/checkpoint/ems/2020_multimodal_blender/q_function/for_fast_acute/results_20200904_153005_live_dialogpt.json \
mephisto.task.task_name=turn_annotations_static_foo

# In-flight QA task, with annotation indices specified:
python parlai/crowdsourcing/tasks/turn_annotations_static/run_in_flight_qa.py \
mephisto.blueprint.data_jsonl=/checkpoint/ems/2020_multimodal_blender/q_function/for_fast_acute/results_20200904_153005_live_dialogpt.json._first4_only \
mephisto.task.task_name=turn_annotations_static_foo \
mephisto.blueprint.annotation_indices_jsonl=/private/home/ems/GitHub/facebookresearch/ParlAI_dev/parlai/crowdsourcing/tasks/turn_annotations_static/task_config/annotation_indices_example.jsonl
```

### CODE REVIEW

**Review by JackUrb — APPROVED** [review]
I love the full readme with details on how to run mephisto tasks using hydra! Also +1 for standardizing the script names.

Lots of quality abstracting in here. LGTM!

File: parlai/crowdsourcing/tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+By default, HITs run locally in sandbox mode. To run live HITs, add `mephisto.provider.requester_name=${REQUESTER_NAME} mephisto/architect=heroku` to your launch command, where `${REQUESTER_NAME}` is the MTurk requester name that you specified when setting up Mephisto.
+ 
+## Saving data
+
+By default, Mephisto data is saved in the following directory:
+```
+<datapath>/data/runs/NO_PROJECT/<project_id>/<task_run_id>/<assignment_id>/<agent_id>/data
+```
+- By default, `<datapath>` is `/scratch/${USER}/mephisto/data`.
```
[the remark this answers is not in the document]
Comment (EricMichaelSmith) [reply parlai/crowdsourcing/tasks/README.md]: Okay cool - I've re-worded this and I hope it's clearer now

File: parlai/crowdsourcing/tasks/turn_annotations_static/run.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (EricMichaelSmith) [reply parlai/crowdsourcing/tasks/turn_annotations_static/run.py:1]: Good point, added

**Comment by EricMichaelSmith** [comment]
@JackUrb @mwillwork @jxmsML I've just made some changes to this, given Mary's new code that offers more control over which turns for each conversation get annotated. Lemme know if you have any more feedback given these changes - otherwise I'll merge this in on Wednesday, since Jack has already approved :)

**Pull request merged**